backend/dataall/core/config.py

```diff
@@ -0,0 +1,44 @@
+"""Reads and encapsulates the configuration provided in config.json"""
+import json
+import copy
+from typing import Any, Dict
+import os
+from pathlib import Path
+
+
+class _Config:
+    """A container of properties in the configuration file
+     and any other that can be specified/overwritten later in the application"""
+
+    def __init__(self):
+        self._config = _Config._read_config_file()
+
+    def get_property(self, key: str) -> Any:
+        """Retrieves a copy of the property"""
+        if key not in self._config:
+            raise KeyError(f"Couldn't find a property {key} in the config")
+
+        return copy.deepcopy(self._config[key])
+
+    def set_property(self, key: str, value: Any) -> None:
+        """Sets a property into the config"""
+        self._config[key] = value
+
+    @staticmethod
+    def _read_config_file() -> Dict[str, Any]:
```

Comment:
Could you elaborate a bit about benefits of storting the config in the json file?

I am personally not very convinced about this approach, I see like this file is added to the container. If at some point we switch to using 'zip' files for lambda functions we will be adding this file to the zip file as well?

I would suggest to consider using any AWS service for storing the config file which were designed for it, did you consider https://docs.aws.amazon.com/appconfig/latest/userguide/what-is-appconfig.html or it doesn't fit into the requirements in here?